### PR TITLE
Allow cookies to always have the Secure attribute if configured

### LINKED ISSUE #58: Using Secure Cookie flag when using 'Authorization Code Flow'
`TransientCookieStore` creates cookie `com.auth0.state`, but while creating this cookie Secure flag is set only if `SameSite` is `None`. But `SameSite` is always set to `Lax` in `AuthorizeUrl` build function when we use default `AuthorizationCodeFlow`. 

Is there a way to use `Secure` flag while still using default `AuthorizationCodeFlow`? 

We are running some vulnerability scans and one of the issue being highlighted is that cooking `com.auth0.state` is not secure (Testing using: User-Agent: Mozilla/5.0 (Windows NT 10.0; WOW64; rv:56.0) Gecko/20100101 Firefox/56.0). 

Apologies if my understanding is not correct, we are still in the process of learning OAuth. Highly appreciate your help to point us in right direction.

### PR DESCRIPTION
**Pull request by jimmyjames**
### Changes

Currently, cookies used for the `state` and `nonce` during the authentication flow only set the `Secure` attribute if `SameSite=None` (this happens if the response type includes `id_token`). This change allows configuration to always set the `Secure` attribute, regardless of the SameSite value.

### References

Closes #58 

### Testing

- [X] This change adds test coverage
- [ ] This change has been tested on the latest version of Java or why not

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [X] All existing and new tests complete without errors

### CODE REVIEW

File: src/main/java/com/auth0/AuthorizeUrl.java [hunk 1]
```diff
@@ -62,6 +63,23 @@ public AuthorizeUrl withConnection(String connection) {
         return this;
     }
 
+    /**
+     * Sets whether cookies used during the authentication flow have the {@code Secure} attribute set or not.
+     * By default, cookies will be set with the Secure attribute if the responseType includes {@code id_token} and thus requires
+     * the {@code SameSite=None} cookie attribute to set. Setting this to false will <strong>not</strong> override this behavior,
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/AuthorizeUrl.java:69]: It reads a bit odd. What about the following?
```suggestion
     * the {@code SameSite=None} cookie attribute set. Setting this to false will <strong>not</strong> override this behavior,
```

File: src/test/java/com/auth0/AuthorizeUrlTest.java [hunk 2]
```diff
@@ -128,6 +128,19 @@ public void shouldSetStateSameSiteAndNotLegacyCookieWhenConfigured() {
         assertThat(headers, hasItem("com.auth0.state=asdfghjkl; HttpOnly; Max-Age=600; SameSite=None; Secure"));
     }
 
+    @Test
+    public void shouldSetSecureCookieWhenConfigured() {
+        String url = new AuthorizeUrl(client, request, response, "https://redirect.to/me", "code")
+                .withState("asdfghjkl")
+                .withSecureCookie(true)
```
Comment (lbalmaceda) [inline src/test/java/com/auth0/AuthorizeUrlTest.java:135]: I'd add a test with this set to false, checking that when samesite=none this is still being set as secure even if we tell it not to with the setter. (the default behavior)

Comment (jimmyjames) [reply src/test/java/com/auth0/AuthorizeUrlTest.java:135]: Pushed a commit to add this test case 👍
